### PR TITLE
Add client credential grant token request

### PR DESCRIPTION
**Pull request by knmeiss**
DDOC-373

## Checklist

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my own changes
- [X ] I have run `yarn lint` to make sure my changes pass all linters
- [X ] I have pulled the latest changes from the upstream developer branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

**Comment by knmeiss** [comment]
`  box_subject_id:
    type: string
    example: '123456789'
    description: |-
      Used in combination with `client_credentials` as the `grant_type`.
      Value is determined by `box_subject_type`.
    enum:
      - enterprise ID
      - user ID`

This is the part causing the issue -- b/c the example doesn't match either enum. I need to tell them the options without enums I guess.

File: content/requests/post_oauth2_token.yml [hunk 1]
```diff
@@ -135,3 +136,22 @@ properties:
...
+
+  box_subject_id:
+    type: string
+    example: '123456789'
+    description: |-
+      Used in combination with `client_credentials` as the `grant_type`.
+      Value is determined by `box_subject_type`.
+    enum:
+      - enterprise ID
+      - user ID
```
Comment (box-devrel) [inline content/requests/post_oauth2_token.yml:157]: @knmeiss So an `enum` is used when there is a finite list of known options for a value, for example a list of template operations https://github.com/box/box-openapi/blob/main/content/attributes/metadata_template_operation.yml#L17-L30

In this case, the value can be any enterprise ID or user ID, and although this is finite it's hardly known or static. I'd just remove these lines to be honest, and instead I'd add something to the description to explain that the subject ID can be an enterprise ID or user ID.

Comment (cbetta) [reply content/requests/post_oauth2_token.yml:157]: @knmeiss So an enum is used when there is a finite list of known options for a value, for example a list of template operations https://github.com/box/box-openapi/blob/main/content/attributes/metadata_template_operation.yml#L17-L30

In this case, the value can be any enterprise ID or user ID, and although this is finite it's hardly known or static. I'd just remove these lines to be honest, and instead I'd add something to the description to explain that the subject ID can be an enterprise ID or user ID.

Comment (cbetta) [inline content/requests/post_oauth2_token.yml:157]: @knmeiss So an enum is used when there is a finite list of known options for a value, for example a list of template operations https://github.com/box/box-openapi/blob/main/content/attributes/metadata_template_operation.yml#L17-L30

In this case, the value can be any enterprise ID or user ID, and although this is finite it's hardly known or static. I'd just remove these lines to be honest, and instead I'd add something to the description to explain that the subject ID can be an enterprise ID or user ID.

File: content/requests/post_oauth2_token.yml [hunk 2]
```diff
@@ -135,3 +136,22 @@ properties:
...
     example: https://api.box.com/2.0/files/123456
+
+  box_subject_type:
+    type: string
+    example: enterprise
+    description: |-
+      Used in combination with `client_credentials` as the `grant_type`.
+    enum:
+      - enterprise
+      - user
```
Comment (cbetta) [inline content/requests/post_oauth2_token.yml:147]: In this case the enum is used correctly.

**Review by box-devrel — APPROVED** [review]

**Pull request merged**